packages/agents/src/functions/RunTestPythonAnalyser.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  Agent,
+  AgentFunctionResult,
+  WrapClient,
+  LlmApi,
+  ChatLogs,
+  Tokenizer,
+  ChatMessageBuilder,
+} from "@evo-ninja/agent-utils";
+import { AgentFunctionBase } from "../AgentFunctionBase";
+import { AgentBaseContext } from "../AgentBase";
+
+interface FunctionParams {
+  filename: string;
+  implementationCode: string;
+}
+
+export class RunTestPythonAnalyser extends AgentFunctionBase<FunctionParams> {
+  constructor(
+    private _llm: LlmApi,
+    private _tokenizer: Tokenizer,
+    private client: WrapClient
+  ) {
+    super();
+  }
+
+  get name() {
+    return "runPythonTest";
+  }
+
+  get description() {
+    return `Run python test and analyses error.`;
+  }
+
+  get parameters() {
+    return {
+      type: "object",
+      properties: {
+        filename: {
+          type: "string",
+        },
+        implementationCode: {
+          type: "string"
+        }
+      },
+      required: ["filename", "implementationCode"],
+      additionalProperties: false,
+    };
+  }
+
+  buildExecutor(_: Agent<unknown>, context: AgentBaseContext) {
+    return async (params: FunctionParams, rawParams?: string): Promise<AgentFunctionResult> => {
+      const testResult = await this.client.invoke<
+        { success: true } | { success: false; error: string }
+      >({
+        uri: "plugin/cmd",
+        method: "runPythonTest",
+        args: {
+          filename: params.filename,
+        },
+      });
+
+      if (!testResult.ok) {
+        throw new Error(testResult.error?.message);
+      }
+
+      if (!testResult.value.success) {
+        const message = `Modify the following implementation code in order to fix the error given:
```

Comment:
I think we're grouping 2 things here that should be seperate.

The `RunPythonTest` function should simply run python tests and return the results. It shouldn't also be the one that's modifying the source code, that is done by the developer agent who is specialized in doing this.